src/tools/wasm-reduce.cpp

```diff
@@ -225,7 +225,7 @@ static std::unordered_set<Name> functionsWeTriedToRemove;
 struct Reducer
   : public WalkerPass<PostWalker<Reducer, UnifiedExpressionVisitor<Reducer>>> {
   std::string command, test, working;
-  bool binary, deNan, verbose, debugInfo;
+  bool binary, deNan, verbose, debugInfo, disableFlatten;
```

Comment:
another option might be for `wasm-reduce` to read the feature section, or be able to receive the feature flags, and then if it knows exceptions are enabled then it automatically doesn't flatten. but that doesn't need to block this.

Reply:
Giving wasm-reduce the capability of reading features section sounds like a good idea, but can't wasm-reduce also read wast files too, in which case we need the option?

I also thought about making this accept the whole set of feature flags, but I was not sure there are gonna be any more case than this in which reducer will be restricted in kinds of passes it can use based on features, and hopefully this restriction will be lifted later even for EH, so this will not last. So I was wondering accepting the whole set of feature flags might be an overkill. WDYT?

Reply:
Yeah, good point, maybe it's overkill to do all features. For now this seems fine.